File: track-api/Models/Db/Dataset.cs
namespace track_api.Models.Db
{
    using System;
    using System.Collections.Generic;
    using System.ComponentModel.DataAnnotations;
    using System.ComponentModel.DataAnnotations.Schema;
    using System.Data.Entity.Spatial;

    [Table("Dataset")]
    public partial class Dataset : DbEntity
    {
        public int? UserId { get; set; }

        [ForeignKey("UserId")]
        public virtual User User { get; set; }

        public bool Archived { get; set; }

        public string Label { get; set; }

        [NotMapped]
        public TimeSpan Span { get; set; }

        public bool Private { get; set; }

        public ICollection<Series> Series { get; set; }

        public ICollection<Record> Records { get; set; }
    }
}

File: track-web/src/models/odata/Dataset.ts
import { Record } from './Record';
import { Series } from './Series';
import { Category } from '.';

export class Dataset {
  public Id: number;
  public UserId: number;
  public Private: boolean;
  public Archived: boolean;
  public Label: string;

  public CategoryId?: number;
  public Category: Category;

  public Span: string;

  private records: Record[];
  private series: Series[];

  constructor(userId?: number) {
    this.UserId = userId;
    this.Private = true;

    this.Label = '';
    this.records = [];
    this.series = [];

    this.series.push(Series.Default());
  }

  public get Records() {
    return this.records;
  }

  public set Records(value: Record[]) {
    this.records = value;
  }

  public get Series() {
    return this.series;
  }

  public set Series(value: Series[]) {
    this.series = value;
  }
}